src/benes.rs

```diff
@@ -351,27 +351,26 @@ pub(crate) fn support_gen(s: &mut [Gf; SYS_N], c: &[u8; COND_BYTES]) {
     for i in 0..(1 << GFBITS) {
         a = util::bitrev(i as Gf);
 
-        for j in 0..GFBITS {
-            l[j][i / 8] |= (((a >> j) & 1) << (i % 8)) as u8;
+        for (j, itr_l) in l.iter_mut().enumerate().take(GFBITS) {
+            itr_l[i / 8] |= (((a >> j) & 1) << (i % 8)) as u8;
         }
     }
 
-    for j in 0..GFBITS {
+    for itr_l in l.iter_mut().take(GFBITS) {
         #[cfg(any(feature = "mceliece348864", feature = "mceliece348864f"))]
         {
-            apply_benes(&mut l[j], c, 0);
+            apply_benes(itr_l, c, 0);
         }
         #[cfg(not(any(feature = "mceliece348864", feature = "mceliece348864f")))]
         {
-            apply_benes(&mut l[j], c, 0);
+            apply_benes(itr_l, c, 0);
         }
     }
 
-    for i in 0..SYS_N {
-        s[i] = 0;
```

Comment:
Is `s` 0 beforehand, since now it does not get zeroed below?

Reply:
no, I think it should be explicitly zeroed & I overlooked this, will be fixed